src/StatsdClient/RandomGenerator.cs

```diff
@@ -4,15 +4,26 @@ namespace StatsdClient
 {
     public class RandomGenerator : IRandomGenerator
     {
-        readonly Random _random;
+        [ThreadStatic]
+        static Random _random;
+
+        private static Random Random
+        {
+            get
+            {
+                var random = _random;
+                if (random != null) return random;
+                return _random = new Random(Guid.NewGuid().GetHashCode());

```

Comment:
If it's static and initiated once, do we need the seed? Also this might be cleaner if `_random` is initiated in a static constructor rather than the null check when accessing.